fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ParameterVisitorBatchDAO.java

```diff
@@ -770,12 +607,14 @@ public void visit(ReferenceParmVal rpv) throws FHIRPersistenceException {
             throw new IllegalArgumentException("Invalid reference parameter value. See server log for details.");
         }
 
+        // reference params are never system-level
+        final boolean isSystemParam = false;
         if (refResourceType != null) {
             // Store a token value configured as a reference to another resource
-            rec = new ResourceTokenValueRec(parameterNameId, resourceType, resourceTypeId, logicalResourceId, refResourceType, refLogicalId, refVersion);
+            rec = new ResourceTokenValueRec(parameterNameId, resourceType, resourceTypeId, logicalResourceId,                   refResourceType, refLogicalId, refVersion, this.currentCompositeId, isSystemParam);
```

Comment:
formatting?

Reply:
yeah I do this sometimes to align the args when I have similar calls and I'm tweaking the signature at the same time. Tricky when several parameters are simply strings and you need to make sure you have the correct order. Ada's typing system would make this so much safer! I've taken out the extra spaces now we no longer need them.